File: 0330-patching-array/0330-patching-array.cpp
class Solution {
public:
    int minPatches(vector<int>& nums, int n) {
        long long sum=0, patches=0, index=0;
        while(sum<n){
            if(index<nums.size() && nums[index]<=sum+1) sum+=nums[index++];
            else {
                patches++;
                sum+=sum+1;
            }
        }
        return patches;
    }
};

File: 0330-patching-array/0330-patching-array.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        right=0
        counter=0
        
        for x in nums:
            while n>right and x>right+1:
                right+=right+1
                counter+=1
            right+=x
        
        while n>right:
            right+=right+1
            counter+=1
        return counter